elyra/pipeline/pipeline_constants.py

```diff
@@ -21,3 +21,4 @@
 PIPELINE_META_PROPERTIES = ["name", "description", "runtime"]
 # optional static prefix to be used when generating an object name for object storage
 COS_OBJECT_PREFIX = "cos_object_prefix"
+NOTEBOOK_EXT = "ipynb"
```

Comment:
There is no need to declare this as a constant. Some background information for context that should help clarify why not. Most constants in this file represent property names that various pieces of code use to access information that the [Pipeline class](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/pipeline.py#L331)  or [Operation class](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/pipeline.py#L219) store. These property names might change over time (e.g. `cos_object_prefix` -> `os_object_prefix`) and we wanted to avoid the need to change the code in several locations. The value that represents a Jupyter notebook file extension (`ipynb`) is not going to change over time, hence it's not a pipeline specific constant.

Reply:
> I ran in a couple tests and the describe feature appeared to work as expected! Good start given the limited background information the original issue provided! There are a few things though that we should improve to help us get the output of the command to where it needs to be to properly support automation.
> 
> Starting with some background information: the `describe` command was introduced to enable users to identify dependencies that a pipeline might have. There are two output formats that this command supports: human readable (the default) and machine readable (if `--json` is specified). The machine readable output can be used to automate certain tasks. For example, a user might write a script that extracts the name of all PVCs and runs the appropriate `kubectl get pvc ..` command to verify that these PVCs are present in the Kubernetes cluster where the pipeline will be executed.
> 
> There are a few obstacles that would prevent the user from doing this:
> 
>     * [This behavior was not introduced by this PR.]The machine readable output includes a header, which shouldn't be printed because it renders the JSON output invalid:
>       ```
>       ──────────...
>       Elyra Pipeline details
>       ───────────...
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       The header should not be printed if machine-readable output is produced.
> 
>     * The machine-readable output for mounted volumes might look as follows:
>       ```
>       "mounted_volumes": [
>          "/mnt/vol2=my-pvc2"
>       ]
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       If a user wanted to use this output (see earlier example) to validate the existence of the specified PVC/s (here `my-pvc2`), it would be necessary to extract the name from `/mnt/vol2=my-pvc2`. We should avoid the need to do so by only returning the PVC name. @kiersten-stokes we had a similar  discussion in [Add support for Kubernetes secrets environment variables #2715](https://github.com/elyra-ai/elyra/pull/2715). Should 2722 piggy back on the changes you are making for that one?

Done.
1. Header Banner is conditional now, applicable when --json is not chosen
2. For mounted_volumes, I split by "=" and took second element
installed and ran jq with the output. Looks good
<img width="1765" alt="image" src="https://user-images.githubusercontent.com/9624108/168060596-00be5916-b0ab-42bc-a043-afaf27f24664.png">

Reply:
> There is no need to declare this as a constant. Some background information for context that should help clarify why not. Most constants in this file represent property names that various pieces of code use to access information that the [Pipeline class](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/pipeline.py#L331) or [Operation class](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/pipeline.py#L219) store. These property names might change over time (e.g. `cos_object_prefix` -> `os_object_prefix`) and we wanted to avoid the need to change the code in several locations. The value that represents a Jupyter notebook file extension (`ipynb`) is not going to change over time, hence it's not a pipeline specific constant.

Understood. Done